crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -948,3 +953,62 @@ impl From<StoredEpochInfo> for Epoch {
         }
     }
 }
+
+impl From<&TransactionKind> for TransactionBlockKind {
+    fn from(value: &TransactionKind) -> Self {
+        match value {
+            TransactionKind::ConsensusCommitPrologue(x) => {
+                let consensus = ConsensusCommitPrologueTransaction {
+                    epoch_id: x.epoch,
+                    round: Some(x.round),
+                    timestamp: DateTime::from_ms(x.commit_timestamp_ms as i64),
+                };
+                TransactionBlockKind::ConsensusCommitPrologueTransaction(consensus)
+            }
+            TransactionKind::ChangeEpoch(x) => {
+                let change = ChangeEpochTransaction {
+                    epoch_id: x.epoch,
+                    timestamp: DateTime::from_ms(x.epoch_start_timestamp_ms as i64),
+                    storage_charge: Some(BigInt::from(x.storage_charge)),
+                    computation_charge: Some(BigInt::from(x.computation_charge)),
+                    storage_rebate: Some(BigInt::from(x.storage_rebate)),
+                };
+                TransactionBlockKind::ChangeEpochTransaction(change)
+            }
+            TransactionKind::Genesis(x) => {
+                let genesis = GenesisTransaction {
+                    objects: Some(
+                        x.objects
+                            .clone()
+                            .into_iter()
+                            .map(Object::from)
+                            .collect::<Vec<_>>(),
+                    ),
+                };
+                TransactionBlockKind::GenesisTransaction(genesis)
+            }
+            _ => unimplemented!(),
+        }
+    }
+}
+
+// TODO fix this GenesisObject
+#[allow(unreachable_code)]
+impl From<GenesisObject> for Object {
+    fn from(value: GenesisObject) -> Self {
+        match value {
+            GenesisObject::RawObject { data, owner } => Self {
+                owner: Some(
+                    SuiAddress::from_bytes(owner.get_owner_address().unwrap().to_vec()).unwrap(),
+                ),
+                storage_rebate: None,
+                bcs: None,
+                previous_transaction: None,
+                kind: None,
+                address: SuiAddress::from_bytes(data.id().into_bytes()).unwrap(),
+                version: unimplemented!(),
+                digest: unimplemented!(),
+            },
+        }
+    }
+}
```

Comment:
This is an invalid conversion and one which you won't be able to implement. I suggest changing the schema

Reply:
I was thinking that might be the case. Changing the schema, that sounds scary to me, so we'd need to discuss this.

Reply:
looks like GenesisObject is a subset of Object w/ just the Data and Owner fields. We could leave the other fields to None for now if we're trying to minimize the number of graphql types

Reply:
Changed the schema to match what we have today in the JSON-RPC. We can modify it again and create a new graphql type, if this is not sufficient.